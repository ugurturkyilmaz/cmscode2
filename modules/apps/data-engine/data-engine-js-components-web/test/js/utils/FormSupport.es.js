/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import * as FormSupport from '../../../src/main/resources/META-INF/resources/js/utils/FormSupport.es';
import createElement from '../__mock__/createElement.es';
import mockPages from '../__mock__/mockPages.es';

let pages = null;

describe('FormSupport', () => {
	beforeEach(() => {
		pages = JSON.parse(JSON.stringify(mockPages));
	});

	afterEach(() => {
		pages = null;
	});

	it('add a new row to the pages and reorder', () => {
		const indexToAddRow = 0;
		const newRow = FormSupport.implAddRow(12, [
			{
				type: 'newRow',
			},
		]);
		const pageIndex = 0;

		expect(
			FormSupport.addRow(pages, indexToAddRow, pageIndex, newRow)
		).toMatchSnapshot();
	});

	it('returns an implementation of a row for the pages', () => {
		const row = [
			{
				spritemap: 'icons.svg',
				type: 'text',
			},
		];
		const size = 12;

		expect(FormSupport.implAddRow(size, row)).toEqual({
			columns: [
				{
					fields: [
						{
							spritemap: 'icons.svg',
							type: 'text',
						},
					],
					size: 12,
				},
			],
		});
	});

	it('gets a specific field through the pages', () => {
		const indexColumn = 0;
		const indexPage = 0;
		const indexRow = 0;

		expect(
			FormSupport.getField(pages, indexPage, indexRow, indexColumn)
		).toMatchSnapshot();
	});

	it('add a new field to column to the pages', () => {
		const columnIndex = 1;
		const field = {
			spritemap: 'icons.svg',
			type: 'text',
		};
		const pageIndex = 0;
		const rowIndex = 0;

		expect(
			FormSupport.addFieldToColumn(
				pages,
				pageIndex,
				rowIndex,
				columnIndex,
				field
			)
		).toMatchSnapshot();
	});

	it('adds a new fields to column void', () => {
		const columnIndex = 2;
		const fields = [
			{
				spritemap: 'icons.svg',
				type: 'text',
			},
		];
		const pageIndex = 0;
		const rowIndex = 1;

		expect(
			FormSupport.setColumnFields(
				pages,
				pageIndex,
				rowIndex,
				columnIndex,
				fields
			)
		).toMatchSnapshot();
	});

	it('removes a column from pages and reorder', () => {
		const columnIndex = 1;
		const pageIndex = 0;
		const rowIndex = 1;

		expect(
			FormSupport.removeColumn(pages, pageIndex, rowIndex, columnIndex)
		).toMatchSnapshot();
	});

	it('removes a fields to column from pages', () => {
		const columnIndex = 1;
		const pageIndex = 0;
		const rowIndex = 1;

		expect(
			FormSupport.removeFields(pages, pageIndex, rowIndex, columnIndex)
		).toMatchSnapshot();
	});

	it('removes a row from pages and reorder', () => {
		const pageIndex = 0;
		const rowIndex = 1;

		expect(
			FormSupport.removeRow(pages, pageIndex, rowIndex)
		).toMatchSnapshot();
	});

	it('gets a column from pages', () => {
		const columnIndex = 1;
		const pageIndex = 0;
		const rowIndex = 1;

		expect(
			FormSupport.getColumn(pages, pageIndex, rowIndex, columnIndex)
		).toMatchSnapshot();
	});

	it('gets a row from pages', () => {
		const pageIndex = 0;
		const rowIndex = 1;

		expect(
			FormSupport.getRow(pages, pageIndex, rowIndex)
		).toMatchSnapshot();
	});

	it('returns true if there are fields in a row', () => {
		const pageIndex = 0;
		const rowIndex = 0;

		expect(
			FormSupport.rowHasFields(pages, pageIndex, rowIndex)
		).toBeTruthy();
	});

	it('returns false if there are fields in a row', () => {
		const pageIndex = 0;
		const rowIndex = 0;

		expect(
			FormSupport.rowHasFields(
				FormSupport.removeFields(pages, pageIndex, rowIndex, 0),
				pageIndex,
				rowIndex
			)
		).toBeFalsy();
	});

	it('extracts the location of the field through the element', () => {
		const element = createElement({
			attributes: [
				{
					key: 'data-ddm-field-column',
					value: 0,
				},
				{
					key: 'data-ddm-field-row',
					value: 2,
				},
				{
					key: 'data-ddm-field-page',
					value: 2,
				},
			],
			tagname: 'div',
		});

		expect(FormSupport.getIndexes(element)).toEqual({
			columnIndex: 0,
			pageIndex: 2,
			rowIndex: 2,
		});
	});

	it('extracts the location of the row through the element', () => {
		const element = createElement({
			attributes: [
				{
					key: 'data-ddm-field-column',
					value: 0,
				},
				{
					key: 'data-ddm-field-row',
					value: 1,
				},
				{
					key: 'data-ddm-field-page',
					value: 2,
				},
			],
			tagname: 'div',
		});

		expect(FormSupport.getIndexes(element)).toEqual({
			columnIndex: 0,
			pageIndex: 2,
			rowIndex: 1,
		});
	});

	it('updates a field', () => {
		const properties = {
			label: 'Foo',
			type: 'radio',
		};

		expect(
			FormSupport.updateField(pages, 'radioField', properties)
		).toMatchSnapshot();
	});
});
