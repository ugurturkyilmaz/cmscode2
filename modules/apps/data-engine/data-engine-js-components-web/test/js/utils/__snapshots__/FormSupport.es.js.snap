// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormSupport add a new field to column to the pages 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [
              Object {
                "spritemap": "icons.svg",
                "type": "text",
              },
            ],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "text1",
                "hideField": true,
                "instanceId": "1fnjndk1",
                "label": "Text Field 1",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
              Object {
                "fieldName": "text2",
                "instanceId": "193jndk1",
                "label": "Text Field 2",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [
              Object {
                "fieldName": "select",
                "instanceId": "1fnjfqk1",
                "label": "Select Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;

exports[`FormSupport add a new row to the pages and reorder 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "type": "newRow",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "text1",
                "hideField": true,
                "instanceId": "1fnjndk1",
                "label": "Text Field 1",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
              Object {
                "fieldName": "text2",
                "instanceId": "193jndk1",
                "label": "Text Field 2",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [
              Object {
                "fieldName": "select",
                "instanceId": "1fnjfqk1",
                "label": "Select Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;

exports[`FormSupport adds a new fields to column void 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "text1",
                "hideField": true,
                "instanceId": "1fnjndk1",
                "label": "Text Field 1",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
              Object {
                "fieldName": "text2",
                "instanceId": "193jndk1",
                "label": "Text Field 2",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [
              Object {
                "fieldName": "select",
                "instanceId": "1fnjfqk1",
                "label": "Select Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [
              Object {
                "spritemap": "icons.svg",
                "type": "text",
              },
            ],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;

exports[`FormSupport gets a column from pages 1`] = `
Object {
  "fields": Array [
    Object {
      "fieldName": "select",
      "instanceId": "1fnjfqk1",
      "label": "Select Field",
      "options": Array [
        Object {
          "label": "Option 1",
          "value": "option1",
        },
        Object {
          "label": "Option 2",
          "value": "option2",
        },
      ],
      "required": true,
      "settingsContext": Object {
        "pages": Array [],
      },
      "type": "select",
    },
  ],
  "size": 6,
}
`;

exports[`FormSupport gets a row from pages 1`] = `
Object {
  "columns": Array [
    Object {
      "fields": Array [
        Object {
          "fieldName": "text1",
          "hideField": true,
          "instanceId": "1fnjndk1",
          "label": "Text Field 1",
          "required": true,
          "settingsContext": Object {
            "pages": Array [],
          },
          "type": "text",
        },
        Object {
          "fieldName": "text2",
          "instanceId": "193jndk1",
          "label": "Text Field 2",
          "required": true,
          "settingsContext": Object {
            "pages": Array [],
          },
          "type": "text",
        },
      ],
      "size": 4,
    },
    Object {
      "fields": Array [
        Object {
          "fieldName": "select",
          "instanceId": "1fnjfqk1",
          "label": "Select Field",
          "options": Array [
            Object {
              "label": "Option 1",
              "value": "option1",
            },
            Object {
              "label": "Option 2",
              "value": "option2",
            },
          ],
          "required": true,
          "settingsContext": Object {
            "pages": Array [],
          },
          "type": "select",
        },
      ],
      "size": 6,
    },
    Object {
      "fields": Array [],
      "size": 2,
    },
  ],
}
`;

exports[`FormSupport gets a specific field through the pages 1`] = `
Object {
  "fieldName": "radio",
  "instanceId": "123njndk1",
  "label": "Radio Field",
  "options": Array [
    Object {
      "label": "Option 1",
      "value": "option1",
    },
    Object {
      "label": "Option 2",
      "value": "option2",
    },
  ],
  "settingsContext": Object {
    "pages": Array [
      Object {
        "rows": Array [
          Object {
            "columns": Array [
              Object {
                "fields": Array [
                  Object {
                    "fieldName": "label",
                    "localizable": true,
                    "settingsContext": Object {
                      "pages": Array [
                        Object {
                          "rows": Array [
                            Object {
                              "columns": Array [],
                            },
                          ],
                        },
                      ],
                    },
                  },
                  Object {
                    "fieldName": "name",
                  },
                  Object {
                    "fieldName": "required",
                  },
                  Object {
                    "fieldName": "type",
                  },
                ],
                "size": 12,
              },
            ],
          },
        ],
      },
    ],
  },
  "type": "radio",
}
`;

exports[`FormSupport removes a column from pages and reorder 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "text1",
                "hideField": true,
                "instanceId": "1fnjndk1",
                "label": "Text Field 1",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
              Object {
                "fieldName": "text2",
                "instanceId": "193jndk1",
                "label": "Text Field 2",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;

exports[`FormSupport removes a fields to column from pages 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "text1",
                "hideField": true,
                "instanceId": "1fnjndk1",
                "label": "Text Field 1",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
              Object {
                "fieldName": "text2",
                "instanceId": "193jndk1",
                "label": "Text Field 2",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;

exports[`FormSupport removes a row from pages and reorder 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;

exports[`FormSupport updates a field 1`] = `
Array [
  Object {
    "description": "Add a short description for this page.",
    "localizedDescription": Object {
      "en_US": "Add a short description for this page.",
    },
    "localizedTitle": Object {
      "en_US": "Page title",
    },
    "rows": Array [
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "radio",
                "instanceId": "123njndk1",
                "label": "Radio Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "settingsContext": Object {
                  "pages": Array [
                    Object {
                      "rows": Array [
                        Object {
                          "columns": Array [
                            Object {
                              "fields": Array [
                                Object {
                                  "fieldName": "label",
                                  "localizable": true,
                                  "settingsContext": Object {
                                    "pages": Array [
                                      Object {
                                        "rows": Array [
                                          Object {
                                            "columns": Array [],
                                          },
                                        ],
                                      },
                                    ],
                                  },
                                },
                                Object {
                                  "fieldName": "name",
                                },
                                Object {
                                  "fieldName": "required",
                                },
                                Object {
                                  "fieldName": "type",
                                },
                              ],
                              "size": 12,
                            },
                          ],
                        },
                      ],
                    },
                  ],
                },
                "type": "radio",
              },
            ],
            "size": 3,
          },
          Object {
            "fields": Array [],
            "size": 9,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "text1",
                "hideField": true,
                "instanceId": "1fnjndk1",
                "label": "Text Field 1",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
              Object {
                "fieldName": "text2",
                "instanceId": "193jndk1",
                "label": "Text Field 2",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "text",
              },
            ],
            "size": 4,
          },
          Object {
            "fields": Array [
              Object {
                "fieldName": "select",
                "instanceId": "1fnjfqk1",
                "label": "Select Field",
                "options": Array [
                  Object {
                    "label": "Option 1",
                    "value": "option1",
                  },
                  Object {
                    "label": "Option 2",
                    "value": "option2",
                  },
                ],
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "select",
              },
            ],
            "size": 6,
          },
          Object {
            "fields": Array [],
            "size": 2,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "date",
                "instanceId": "1ffqwjndk1",
                "label": "Date Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "date",
              },
            ],
            "size": 12,
          },
        ],
      },
      Object {
        "columns": Array [
          Object {
            "fields": Array [
              Object {
                "fieldName": "numeric",
                "instanceId": "1ljqjndk1",
                "label": "Numeric Field",
                "required": true,
                "settingsContext": Object {
                  "pages": Array [],
                },
                "type": "numeric",
              },
            ],
            "size": 12,
          },
        ],
      },
    ],
    "title": "Page title",
  },
]
`;
